internal/nix/source.go

```diff
@@ -26,27 +25,38 @@ func SourceNixEnv() error {
 		)
 	}
 
+	// Source the nix script that sets the environment, and print the environment
+	// variables that it sets, in a way we can parse.
+
+	// NOTE: Only use shell built-ins in this script so that we don't introduce
+	// any dependencies on an external binary.
+	script := heredoc.Docf(`
+		. %s;
+		echo PATH=$PATH;
+		echo NIX_PROFILEs=$NIX_PROFILES;
+		echo NIX_SSL_CERT_FILE=$NIX_SSL_CERT_FILE;
+	`, srcFile)
 	cmd := exec.Command(
 		"/bin/sh",
 		"-c",
-		fmt.Sprintf(". %s ; echo '<<<ENVIRONMENT>>>' ; env", srcFile),
+		script,
 	)
 
 	bs, err := cmd.CombinedOutput()
 	if err != nil {
-		return errors.WithStack(err)
+		// When there's an error, the output is usually an error message that
+		// was printed to stderr and that we want in the error for debugging.
+		return errors.WithMessage(err, string(bs))
```

Comment:
You probably want `Wrap` instead of `WithMessage`. Wrap adds a stack but WithMessage does not.

(tbh, no idea why `WithMessage` exists)

Reply:
Fixed